cli.js

```diff
@@ -1,248 +1,269 @@
 #!/usr/bin/env node
 'use strict';
 
-var debug = require('debug')('ava');
-
-if (debug.enabled) {
-	require('time-require');
-}
-
-var fs = require('fs');
-var path = require('path');
-var figures = require('figures');
-var flatten = require('arr-flatten');
-var globby = require('globby');
-var meow = require('meow');
-var updateNotifier = require('update-notifier');
-var chalk = require('chalk');
-var Promise = require('bluebird');
-var fork = require('./lib/fork');
-var log = require('./lib/logger');
-
-// Bluebird specific
-Promise.longStackTraces();
-
-var cli = meow([
-	'Usage',
-	'  ava [<file|folder|glob> ...]',
-	'',
-	'Options',
-	'  --init       Add AVA to your project',
-	'  --fail-fast  Stop after first test failure',
-	'  --serial     Run tests serially',
-	'',
-	'Examples',
-	'  ava',
-	'  ava test.js test2.js',
-	'  ava test-*.js',
-	'  ava --init',
-	'  ava --init foo.js',
-	'',
-	'Default patterns when no arguments:',
-	'test.js test-*.js test/*.js'
-], {
-	string: ['_'],
-	boolean: [
-		'fail-fast',
-		'serial'
-	]
-});
-
-var rejectionCount = 0;
-var exceptionCount = 0;
-var testCount = 0;
-var fileCount = 0;
-var errors = [];
-
-function prefixTitle(file) {
-	var separator = ' ' + chalk.gray.dim(figures.pointerSmall) + ' ';
-
-	var base = path.dirname(cli.input[0]);
-
-	if (base === '.') {
-		base = cli.input[0] || 'test';
+(function () {
+	// Prefer the local installation of AVA.
+	var resolveFrom = require('resolve-from');
+	var localCLI;
+	try {
```

Comment:
FYI [try-resolve-from](https://github.com/shinnn/try-resolve-from/blob/50cd2cb89934ec0172e43db86fe78263759908ee/index.js) works as you expected. https://github.com/shinnn/try-resolve-from/blob/50cd2cb89934ec0172e43db86fe78263759908ee/index.js